hello-security/login-controller.php

```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <title>Login</title>
+</head>
+<body>
+<a href="index.php"> Take me home, country roads ... </a> <br>
+<?php
+include "db-connection.php";
+$conn = openCon();
+
+$email = htmlentities($_POST["email"]);
+$password = htmlentities($_POST["password"]);
+
+$sql = "SELECT * from person where email = :email";
+
+//prepared statement - defence against SQL injection
+$stmt = $conn->prepare($sql);
+
+$stmt->bindParam(':email', $email);
+
+$result = $stmt->execute() or die("Failed to query from DB!");
+
+$firstrow = $stmt->fetch(PDO::FETCH_ASSOC) or die ("Not valid email.");
+
+if (password_verify($password, $firstrow['password'])) {
+	echo("Hello " . $firstrow['firstname'] . " you are now logged in.");
+	session_start();
+	$_SESSION["email"] = $firstrow['email'];
+} else {
+	die ("Not valid password for the provided email.");
```

Comment:
You may want to change the error. This way currently the user knows that the email he is trying to log in with is existing in the system. This could be considered security sensitive information that we may want to prevent from being shared.